app/models/user.rb

```diff
@@ -18,12 +23,13 @@ class User < ApplicationRecord
   has_many :incoming_potential_swaps, class_name: "PotentialSwap", foreign_key: "target_user_id", dependent: :destroy
   has_many :sent_emails, dependent: :destroy
 
-  delegate :profile_url, to: :identity, prefix: false
-
   before_save :clear_swap, if: :details_changed?
   after_save :send_welcome_email, if: :needs_welcome_email?
   before_destroy :clear_swap
 
+  validates :email, uniqueness: { case_sensitive: false }, allow_nil: true
+  validates :name, presence: true
```

Comment:
While I love validators, this could cause mayhem against any old data we have floating around with blank names, are we sure there is no old data with nameless users?